### PR TITLE
feat: add JWT auth, error handling

### PR DESCRIPTION
**Pull request by jkerola**
- add auth routes for register, login, profile
- add DTOs with basic property validation
- add jwt creation and handling
- add @requires_authentication decorator for routes 
- error handler will catch any workzeug.exceptions.HTTPException raised and return it as JSON

### CODE REVIEW

File: api/models/auth_dtos.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            email=data.get("email"),
+            firstName=data.get("firstName"),
+            lastName=data.get("lastName"),
+        )
+
+
+@dataclass(frozen=True)
+class LoginDto(_BaseDto):
+    username: str
+    password: str
```
Comment (emipyy) [inline api/models/auth_dtos.py:58]: Do we plan to store passwords as hashes ever?

**Review by emipyy — APPROVED** [review]
Some small questions, otherwise looks good. It may be good to add example `.env` file for the `load_dotenv()` method. Also, a chart visualizing the auth flow could be good.

File: .pre-commit-config.yaml
[the remark this answers is not in the document]
Comment (jkerola) [reply .pre-commit-config.yaml:1]: I've been working on multiple machines, each with a slightly different version of python3. This hook wouldn't pass unless the local version matches the config version.

I don't think we need to force a specific version of python3 for development, and by removing this bit, pre-commit will default to the locally installed python version. The original 3.11 was just something I included by default without thinking about it.

File: api/models/auth_dtos.py (hunk 1 above)
Comment (jkerola) [reply api/models/auth_dtos.py:58]: I will update the readme and create a small chart to help visualize this entire thing. Thanks for the review!

**Pull request merged**